typing/typecore.ml

```diff
@@ -2913,6 +2935,8 @@ and type_expect_
     end
   | Pexp_constant cst ->
       let cst = constant_or_raise env loc cst in
+      if is_allocated_constant cst then
+        register_allocation expected_mode;
```

Comment:
Is there any particular reason to want constants to be in mode Local?

Reply:
Probably not, but I'd rather just treat all allocations uniformly than reason about which ones are guaranteed to be statically allocated, at least whilst the cost is insignificant.

(I suppose that if you use `-unsafe-string` then it still matters for strings, but I don't think anyone cares about that)

Reply:
`-unsafe-string` isn't even an option any more, FWIW.